.golangci.yml

```diff
@@ -0,0 +1,117 @@
+run:
+  timeout: 3m
+  build-tags:
+    - '!wasm'
+
+linters-settings:
+  gofmt:
+    simplify: false
+  govet:
+    check-shadowing: false # TODO enable
+  nolintlint:
+    allow-unused: false # report any unused nolint directives
+    require-explanation: true # require an explanation for nolint directives
+    require-specific: true # require nolint directives to mention the specific linter being suppressed
+  gocyclo:
+    min-complexity: 20
+  goconst:
+    ignore-tests: true
+  goheader:
+    template-path: '.golangci.goheader.template'
+    values:
+      regexp:
+        copyright-year: 20[2-9]\d
+
+issues:
+  exclude-rules:
+    - path: '(.+)acceptance_testing\.go'
+      linters:
+        - stylecheck
+      text: "ST1003: should not use underscores in Go names"
+    # Exclude some linters from running on tests files.
+    - path: _test\.go
+      linters:
+        - dogsled
+        - gosec
+
+linters:
+  # please, do not use `enable-all`: it's deprecated and will be removed soon.
+  # inverted configuration with `enable-all` and `disable` is not scalable during updates of golangci-lint
+  disable-all: true
+  enable:
+    # We plan to enable all of the linters which are commented out.
+    # However, we want to enable them one by one (so we don't have to fix many issues at once).
```

Comment:
That's from Conduit, but given that we're starting this project from scratch I think we can be a bit stricter. A good start IMO would be the config we have in [conduit-commons](https://github.com/ConduitIO/conduit-commons/blob/main/.golangci.yml).

Reply:
Done!